packages/mattermost-redux/src/reducers/entities/threads/threadsInTeam.ts

```diff
@@ -5,11 +5,22 @@ import {ChannelTypes, PostTypes, TeamTypes, ThreadTypes, UserTypes} from 'matter
 import {GenericAction} from 'mattermost-redux/types/actions';
 import {Team} from 'mattermost-redux/types/teams';
 import {ThreadsState, UserThread} from 'mattermost-redux/types/threads';
+import type {$ID, IDMappedObjects} from 'mattermost-redux/types/utilities';
 
 import {ExtraData} from './types';
 
 type State = ThreadsState['threadsInTeam'] | ThreadsState['unreadThreadsInTeam'];
 
+// return true only if it's 'newer' than other threads
+// older threads will be added by scrolling so no need to manually add.
+// furthermore manually adding older thread will BREAK pagination
+function shouldAddThreadId(ids: Array<$ID<UserThread>>, thread: UserThread, threads: IDMappedObjects<UserThread>) {
+    return ids.some((id) => {
```

Comment:
Is it intended that this returns true if the given thread is newer than at least one of the stored threads? Perhaps it should only be added if it's newer than every stored thread? 0/5 on whether that should be the case though.

Reply:
The case is a bit weird:
We load threads from newest to oldest (top to bottom) in the list, 
and then paginate asking each time the threads before the bottom one, 
older threads will be shown in the list upon scrolling so no problem there.
if we manually add them they might create gaps in the pagination so we don't*. 

Newer threads (or in-between threads) though, would never have a change to be added via scrolling
since we already are past that scrolling point (that's why they are newer), 
so each thread newer than the oldest one should be manually added in the list or 
we won't have a chance to load them**.

Reply:
Ah, I see now. That makes sense then.

Not something to change here, but this seems like a very similar problem to how we have to keep track of posts that are loaded in a sparse list where there might be items missing. That code is very post-specific at the moment, and there's some changes I'd like to make to simplify it if we end up ever needing to update it, but I'm wondering now if there's anything that we could do to make it more generic. At the time, it was the only place where we cared about loading part of a list of items that are sorted by time, but loading threads for that list does feel very similar. But at the same time, threads can change order which makes that messier, although thankfully don't have to be loaded sparsely since you're always viewing them in recency order.

Anyway, that's just some rambling thoughts on my part. This looks good to me 😛